### ISSUE TITLE
Some wrong in function detector_cleanse()

### ISSUE THREAD

**Issue by zBazinga** [jeongjin0/detector-cleanse#4]
When modifying the code, I noticed that on line 81 of the detector_cleanse function, we get the probs by `H_sum += calculate_entropy(probs[0][max_index].clone().detach())`,but the probs is the output of img without clean fature; 

However, according to the algorithm in the paper, it might be necessary to obtain the predicted probability after adding the clean feature.

**Comment by jeongjin0** [comment]
Thank you for bringing this to our attention. You're absolutely correct, and we appreciate your keen observation. 

We've modified the code as you suggested to align it more closely with the algorithm described in the paper. The updated code now looks like this:

```python
perturbed_prediction = model.predict([perturbed_image], [img.shape[1:]])

...

perturbed_probs = perturbed_prediction[3]
H_sum += calculate_entropy(perturbed_probs[0][max_index].clone().detach())
```

This change aligns the implementation more closely with the algorithm described in the paper. Thank you for your valuable contribution to improving our code.
